File: 1.two-sum.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        int_dict = {}
        for index, num in enumerate(nums):
            if target - num in int_dict:
                return [int_dict[target - num], index]
            else:
                int_dict[num] = index

File: 1.two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> int_map;
        for (int i = 0; i < nums.size(); i++) {
            auto iter = int_map.find(target - nums[i]);
            if (iter == int_map.end()) {
                int_map[nums[i]] = i;
            } else {
                return vector<int>{iter->second, i};
            }
        }
    }
};